.github/workflows/deploy-python-backend-dev.yml

```diff
@@ -0,0 +1,36 @@
+name: Deploy the frontend and backend to the Firebase Production environment.
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - llm/**
+      - firestore.rules
+      - firestore.indexes.json
+      - firebase.json
+      - storage.rules
+
+jobs:
+  build_and_deploy:
+    # Don't deploy forks, even if actions are enabled
+    if: github.repository_owner == 'codeforboston'
+    runs-on: ubuntu-latest
+    environment: dev
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+      - name: Setup Python environment
+        uses: ./.github/actions/python
+      - run: |
+          python -m venv llm/venv
+          source llm/venv/bin/activate
+          pip3 install -r llm/requirements.txt
```

Comment:
I don't yet know if this will actually work. It is unclear what is available from the install python action.